client/src/components/ResetPasswordPage/ResetPasswordPage.jsx

```diff
@@ -0,0 +1,122 @@
+import styles from "./ResetPasswordPage.module.css";
+import Paper from "@mui/material/Paper";
+import Typography from "@mui/material/Typography";
+import Grid from "@mui/material/Grid";
+import { useState } from "react";
+import { Button, Input, TextField } from "@mui/material";
+import queryString from "query-string";
+
+function ResetPasswordPage() {
+  const parsed = queryString.parse(location.search);
+  console.log(parsed);
+  console.log(parsed.token);
+  const [newPassword, setNewPassword] = useState("");
+  const [newPasswordVerify, setNewPasswordVerify] = useState("");
+  const fetchResetPasswordApi = async () => {
+    await fetch(
+      `http://localhost:5000/auth/reset-password?token=${parsed.token}`,
+      {
+        method: "POST",
+        headers: {
+          Authorization: "Bearer " + parsed.token,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ password: newPassword }),
+      }
+    ).then((response) => {
+      console.log(response);
+      if (response.status === 401) {
+        alert("Invalid or expired password reset token");
+      } else if (response.status === 200) {
+        alert(
+          "Password reset successful. Please use the new password to log in."
+        );
+      }
+    });
+  };
+  const onSubmit = (event) => {
+    event.preventDefault();
+    if (newPassword !== newPasswordVerify) {
+      alert("Passwords do not match");
+    } else {
```

Comment:
Thank you for the detailed advise! It is very helpful for correction. 

- As advised, **else** is inserted before fetchResetPasswordApi. 
- Conflict in **App.jsx** has been resolved.